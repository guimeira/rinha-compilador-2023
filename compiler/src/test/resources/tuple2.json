{"name":"tuple2.rinha","expression":{"kind":"Let","name":{"text":"f1","location":{"start":4,"end":6,"filename":"tuple2.rinha"}},"value":{"kind":"Function","parameters":[{"text":"c","location":{"start":13,"end":14,"filename":"tuple2.rinha"}}],"value":{"kind":"Let","name":{"text":"e","location":{"start":27,"end":28,"filename":"tuple2.rinha"}},"value":{"kind":"Function","parameters":[],"value":{"kind":"Str","value":"e","location":{"start":40,"end":43,"filename":"tuple2.rinha"}},"location":{"start":31,"end":43,"filename":"tuple2.rinha"}},"next":{"kind":"Let","name":{"text":"f2","location":{"start":51,"end":53,"filename":"tuple2.rinha"}},"value":{"kind":"Tuple","first":{"kind":"Str","value":"a","location":{"start":57,"end":60,"filename":"tuple2.rinha"}},"second":{"kind":"Str","value":"b","location":{"start":62,"end":65,"filename":"tuple2.rinha"}},"location":{"start":56,"end":66,"filename":"tuple2.rinha"}},"next":{"kind":"Function","parameters":[{"text":"d","location":{"start":76,"end":77,"filename":"tuple2.rinha"}}],"value":{"kind":"Tuple","first":{"kind":"Var","text":"f2","location":{"start":89,"end":91,"filename":"tuple2.rinha"}},"second":{"kind":"Tuple","first":{"kind":"Var","text":"c","location":{"start":94,"end":95,"filename":"tuple2.rinha"}},"second":{"kind":"Tuple","first":{"kind":"Var","text":"d","location":{"start":98,"end":99,"filename":"tuple2.rinha"}},"second":{"kind":"Binary","lhs":{"kind":"Call","callee":{"kind":"Var","text":"e","location":{"start":101,"end":102,"filename":"tuple2.rinha"}},"arguments":[],"location":{"start":101,"end":104,"filename":"tuple2.rinha"}},"op":"Add","rhs":{"kind":"Str","value":"fg","location":{"start":107,"end":111,"filename":"tuple2.rinha"}},"location":{"start":101,"end":111,"filename":"tuple2.rinha"}},"location":{"start":97,"end":112,"filename":"tuple2.rinha"}},"location":{"start":93,"end":113,"filename":"tuple2.rinha"}},"location":{"start":88,"end":114,"filename":"tuple2.rinha"}},"location":{"start":72,"end":118,"filename":"tuple2.rinha"}},"location":{"start":47,"end":118,"filename":"tuple2.rinha"}},"location":{"start":23,"end":118,"filename":"tuple2.rinha"}},"location":{"start":9,"end":120,"filename":"tuple2.rinha"}},"next":{"kind":"Let","name":{"text":"f3","location":{"start":127,"end":129,"filename":"tuple2.rinha"}},"value":{"kind":"Call","callee":{"kind":"Var","text":"f1","location":{"start":132,"end":134,"filename":"tuple2.rinha"}},"arguments":[{"kind":"Str","value":"c","location":{"start":135,"end":138,"filename":"tuple2.rinha"}}],"location":{"start":132,"end":139,"filename":"tuple2.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":145,"end":146,"filename":"tuple2.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"First","value":{"kind":"Call","callee":{"kind":"Var","text":"f3","location":{"start":167,"end":169,"filename":"tuple2.rinha"}},"arguments":[{"kind":"Str","value":"d","location":{"start":170,"end":173,"filename":"tuple2.rinha"}}],"location":{"start":167,"end":174,"filename":"tuple2.rinha"}},"location":{"start":161,"end":175,"filename":"tuple2.rinha"}},"location":{"start":155,"end":176,"filename":"tuple2.rinha"}},"location":{"start":149,"end":177,"filename":"tuple2.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":183,"end":184,"filename":"tuple2.rinha"}},"value":{"kind":"Print","value":{"kind":"Second","value":{"kind":"First","value":{"kind":"Call","callee":{"kind":"Var","text":"f3","location":{"start":206,"end":208,"filename":"tuple2.rinha"}},"arguments":[{"kind":"Str","value":"d","location":{"start":209,"end":212,"filename":"tuple2.rinha"}}],"location":{"start":206,"end":213,"filename":"tuple2.rinha"}},"location":{"start":200,"end":214,"filename":"tuple2.rinha"}},"location":{"start":193,"end":215,"filename":"tuple2.rinha"}},"location":{"start":187,"end":216,"filename":"tuple2.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":222,"end":223,"filename":"tuple2.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"Second","value":{"kind":"Call","callee":{"kind":"Var","text":"f3","location":{"start":245,"end":247,"filename":"tuple2.rinha"}},"arguments":[{"kind":"Str","value":"d","location":{"start":248,"end":251,"filename":"tuple2.rinha"}}],"location":{"start":245,"end":252,"filename":"tuple2.rinha"}},"location":{"start":238,"end":253,"filename":"tuple2.rinha"}},"location":{"start":232,"end":254,"filename":"tuple2.rinha"}},"location":{"start":226,"end":255,"filename":"tuple2.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":261,"end":262,"filename":"tuple2.rinha"}},"value":{"kind":"Print","value":{"kind":"First","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"Call","callee":{"kind":"Var","text":"f3","location":{"start":291,"end":293,"filename":"tuple2.rinha"}},"arguments":[{"kind":"Str","value":"d","location":{"start":294,"end":297,"filename":"tuple2.rinha"}}],"location":{"start":291,"end":298,"filename":"tuple2.rinha"}},"location":{"start":284,"end":299,"filename":"tuple2.rinha"}},"location":{"start":277,"end":300,"filename":"tuple2.rinha"}},"location":{"start":271,"end":301,"filename":"tuple2.rinha"}},"location":{"start":265,"end":302,"filename":"tuple2.rinha"}},"next":{"kind":"Print","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"Second","value":{"kind":"Call","callee":{"kind":"Var","text":"f3","location":{"start":331,"end":333,"filename":"tuple2.rinha"}},"arguments":[{"kind":"Str","value":"d","location":{"start":334,"end":337,"filename":"tuple2.rinha"}}],"location":{"start":331,"end":338,"filename":"tuple2.rinha"}},"location":{"start":324,"end":339,"filename":"tuple2.rinha"}},"location":{"start":317,"end":340,"filename":"tuple2.rinha"}},"location":{"start":310,"end":341,"filename":"tuple2.rinha"}},"location":{"start":304,"end":342,"filename":"tuple2.rinha"}},"location":{"start":257,"end":342,"filename":"tuple2.rinha"}},"location":{"start":218,"end":342,"filename":"tuple2.rinha"}},"location":{"start":179,"end":342,"filename":"tuple2.rinha"}},"location":{"start":141,"end":342,"filename":"tuple2.rinha"}},"location":{"start":123,"end":342,"filename":"tuple2.rinha"}},"location":{"start":0,"end":342,"filename":"tuple2.rinha"}},"location":{"start":0,"end":342,"filename":"tuple2.rinha"}}
