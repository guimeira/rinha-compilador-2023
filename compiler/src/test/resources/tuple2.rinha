let f1 = fn (c) => {
  let e = fn () => "e";
  let f2 = ("a", "b"); 

  fn (d) => {
    (f2, (c, (d, e() + "fg")))
  }
};

let f3 = f1("c");
let _ = print(first(first(f3("d"))));
let _ = print(second(first(f3("d"))));
let _ = print(first(second(f3("d"))));
let _ = print(first(second(second(f3("d")))));
print(second(second(second(f3("d")))))
