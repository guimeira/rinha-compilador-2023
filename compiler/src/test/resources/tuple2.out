a
b
c
d
efg
